source/common/tcp_proxy/upstream.h

```diff
@@ -183,17 +183,24 @@ class HttpUpstream : public GenericUpstream, protected Http::StreamCallbacks {
     void decodeData(Buffer::Instance& data, bool end_stream) override {
       parent_.upstream_callbacks_.onUpstreamData(data, end_stream);
       if (end_stream) {
+        done_reading_called_ = true;
```

Comment:
I don't think this boolean is necessary.  We should only get end stream once 
Arguably we should handle done reading in headers as well if there's a headers-only response.

Reply:
Is it not possible that end_stream would be true but decodeTrailers will be called as well? If not then I agree and the boolean is redundant.

Reply:
Also @alyssawilk seems like in decodeHeaders the resetEncoder is being called directly in case of headers-only response (then the end_stream) would be true. Practically doneReading() would call resetEncoder, so this is case is covered. What do you say?